### ISSUE TITLE
P2 🌦️🎲 Let tiny modes notice the weather around them

### ISSUE THREAD

**Issue by cctsao1008** [cctsao1008/coffee-routine-model#17]
# Context-Dependent Mode Transitions 🌦️🎲☕

The hybrid model currently uses a fixed mode-transition matrix.

That is tidy, but a little too forgetful about context. XD

A routine in a long quiet stretch, a coordinated leave, or a recovery period should not necessarily use the exact same transition probabilities.

## Tiny equation 🧠✨

Move from:

```text
p(m[t+1] | m[t])
```

toward:

```text
p(m[t+1] | m[t], x[t], a[t], d[t], regime[t])
```

without making the model so complicated that nobody can inspect it anymore.

## Tiny mission 🌱

- [ ] define a compact context feature vector for transitions
- [ ] preserve the fixed matrix as a reproducible baseline
- [ ] implement inspectable context-conditioned transition logits/probabilities
- [ ] allow recovery evidence to increase plausible Recovery transitions
- [ ] allow coordinated exception/leave context without treating it as failure
- [ ] prevent single noisy clues from swinging mode probabilities wildly
- [ ] compare fixed vs context-aware transition performance on synthetic worlds
- [ ] test transition probabilities always normalize correctly
- [ ] document every transition feature and assumption

## Important tiny rule ☕

```text
Context can change transition probability.
Context does not deterministically choose the mode.
```

Still stochastic. Still uncertain. Still no telepathy. XD

## Done when 🌟

The same previous mode can produce meaningfully different next-mode distributions under different known contexts, and the improvement is measurable against the fixed-matrix baseline.

Tiny weather matters. Tiny dice still roll. 🌦️🎲🐣

**Comment by cctsao1008** [comment]
🌦️🎲 The tiny mode dice can see the weather now.

Implemented:

```text
coffee_brain/transitions.py
tiny_tools/compare_transitions.py
tests/test_transitions.py
docs/transition-weather.md
```

The old fixed `p(m[t+1] | m[t])` matrix stays the default reproducible baseline. Context-aware transitions are explicit opt-in and use bounded log-probability nudges from state, observable actions, explicit disturbance/special/recovery context, and optional regime logits.

Semantic guardrails are baked in:

```text
pass alone != Leave evidence
ordinary delivery + ACK != Recovery evidence
Special needs explicit special-event context
context != deterministic mode choice
```

Validated CI smoke race, 240 synthetic days, seed 20260908:

```text
context-aware mean NLL : 1.0885
fixed-matrix mean NLL  : 1.1819
NLL improvement        : +0.0934
context-aware top-1    : 56.49%
fixed-matrix top-1     : 57.32%
pytest                  : 83 passed
CI                      : PASS
```

So the richer model wins probabilistic fit on the context-generated synthetic sequence, while fixed happens to win top-1 by a tiny amount — a useful reminder that one metric is not model quality. XD
